### PR TITLE
Students program. Angular JS. Task 2

### PR DESCRIPTION
**Pull request by digitalmaxing**

### CODE REVIEW

File: script/controller/tasklist-front.js [hunk 1]
```diff
@@ -89,4 +103,57 @@ angular.module('tasklist-front', ['tasklist-back', 'utils'])
...
+    .directive('userExist', ['Users', function (Users) {
+        return {
+            require: '?ngModel',
+            link: function (scope, elem, attrs, ctrl) {
+                scope.$watch(elem.attr('ng-model'), function (user) {
+                    ctrl.$setValidity('userExist', Users.isExistent(user));
+                });
+            }
+        }
+    }])
```
Comment (kobylianskiy) [inline script/controller/tasklist-front.js:142]: A good practice is to use unique prefix for your directive names. It allows you to determine the origin of the directive, e.g ng prefix is used for core angular directives.

File: views/task-items.html [hunk 2]
```diff
@@ -9,28 +10,32 @@
...
 
-            <input type="submit" class="btn save-btn" ng-click="saveDescription(item); item.focused = false;"
-                   value="Save"
-                   ng-disabled="taskItemForm.description.$invalid" ng-show="item.access.edit && item.focused">
             <input type="submit" class="btn cancel-btn" ng-click="restoreDescription(item); item.focused = false;"
                    value="Cancel"
                    ng-show="item.access.edit && item.focused">
+            <input type="submit" class="btn save-btn" ng-click="saveDescription(item); item.focused = false;"
+                   value="Save"
+                   ng-show="item.access.edit && item.focused">
```
Comment (kobylianskiy) [inline views/task-items.html:32]: Your model shouldn't know about ui related things as focused and hovered.

File: views/tasks-widget.html [hunk 3]
```diff
@@ -2,18 +2,18 @@
...
 
-            <input type="submit" class="btn new-task-btn" value="New task"
-                   ng-disabled="newTaskForm.description.$invalid">
+            <input type="submit" class="btn new-task-btn" value="New task">
         </div>
     </form>
 
     <div class="widget-properties">
-        <input type="checkbox" id="only-assigned-checkbox" ng-model="onlyAssigned" ng-change="filterAssigned();"/>
+        <input type="checkbox" id="only-assigned-checkbox" ng-model="onlyAssigned"/>
```
Comment (kobylianskiy) [inline views/tasks-widget.html:16]: Maybe  'onlyAssigned' can be replaced with some better naming.

File: views/task-items.html (hunk 2 above)
Comment (digitalmaxing) [inline views/task-items.html:32]: My model (I mean "Tasks") actually doesn't know about things like "hovered" and "focused". It returns only tasks and accesses.
And the "widgetController" appends this attributes to the data returned by model.
I want fully understand why in this case it's bad and after that make a changes.

File: views/tasks-widget.html (hunk 3 above)
Comment (digitalmaxing) [inline views/tasks-widget.html:16]: Agreed

File: script/controller/tasklist-front.js (hunk 1 above)
Comment (digitalmaxing) [inline script/controller/tasklist-front.js:142]: Not bad idea. I'll use

File: script/controller/tasklist-front.js [hunk 4]
```diff
@@ -89,4 +103,57 @@ angular.module('tasklist-front', ['tasklist-back', 'utils'])
...
+                        break;
+                    case Status.FINISHED :
+                        _finished.push(item);
+                        break;
+                }
+            });
+
+            return _reopened.concat(_new, _finished);
+        }
+    })
```
Comment (dkalinin) [inline script/controller/tasklist-front.js:131]: Any sorting by date? A better solution is to use .sort() with proper compare function.

File: script/service/users-back.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @author Bogdan Kovalev
+ */
+
+angular.module('users-back', [])
+    .factory('Users', function () {
+
+        function Users() {
+            var currentUser = "Bogdan";
```
Comment (dkalinin) [inline script/service/users-back.js:9]: Bad idea when backend decides who is current user. User can be set by ng-init in front and then it should be passed to backend if needed.

Comment (digitalmaxing) [inline script/service/users-back.js:9]: Agreed...
I'll change during the login page implementation.

**Pull request merged**